app/authentication/session_storage.py

```diff
@@ -1,144 +1,112 @@
-from uuid import uuid4
-
-from flask import session
 from structlog import get_logger
-
 from app.data_model.models import EQSession, db
+from app.data_model.session_data import SessionData
+from app.settings import EQ_SESSION_ID
+from flask import session as cookie_session
+import simplejson as json
 
-USER_ID = 'user_id'
-USER_IK = 'user_ik'
-SURVEY_COMPLETED_METADATA = 'survey_completed_metadata'
-EQ_SESSION_ID = 'eq-session-id'
 
 logger = get_logger()
 
 
 class SessionStorage:
 
-    @staticmethod
-    def store_user_id(user_id):
+    def __init__(self):
+        self.eq_session = None
+
+    def store_session_variables(self, user_id, eq_session_id, metadata):
         """
-        Create a new eq_session_id and associate it with the user_id specified
+        Create a new eq_session and associate it with the user_id specified
         """
-        eq_session_id = str(uuid4())
-        session[EQ_SESSION_ID] = eq_session_id
-        eq_session = EQSession(eq_session_id, user_id)
+        session_data = self._get_session_data(metadata)
+        self.eq_session = EQSession(eq_session_id, user_id, session_data)
 
         logger.debug('Adding eq_session to database', eq_session_id=eq_session_id, user_id=user_id)
         # pylint: disable=maybe-no-member
-        db.session.add(eq_session)
+        db.session.add(self.eq_session)
         db.session.commit()
 
+    @staticmethod
+    def _get_session_data(metadata):
+        session_data = SessionData(
+            tx_id=metadata.get('tx_id'),
+            eq_id=metadata.get('eq_id'),
+            form_type=metadata.get('form_type'),
+            ru_ref=metadata.get('ru_ref'),
+            ru_name=metadata.get('ru_name'),
+            period_str=metadata.get('period_str'),
+        )
+        return json.dumps(vars(session_data))
+
+    def add_submission_time(self):
+        """
+        add submission time to session data
+        """
+        self.eq_session = self._get_user_session()
```

Comment:
we shouldn't need to re-assign `self.eq_session` as it is already assigned by `self._get_user_session()`